### PR TITLE
WT-6204 Possible race between backup and checkpoint at file close

### PR DESCRIPTION
**Pull request by jiechenbo**
This ticket involves researching if there could be a potential race condition between backup and checkpoint at file close. This race condition can only happen if:

- It is not the last checkpoint for the system
- We don't have the schema lock
- We don't have the checkpoint lock 

Thus, upon researching, the assert was never hit. It would be useful to still include this assert to prevent future possible race conditions.

### CODE REVIEW

File: src/txn/txn_ckpt.c [hunk 1]
```diff
@@ -1948,6 +1948,16 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
...
+    /*
+     * Checkpoint shouldn't occur when final is set to false, without either checkpoint lock or
+     * schema lock. Potential race condition arises when backup is ongoing, causing inconsistencies
+     * between the versions of data.
+     */
+    WT_ASSERT(session,
+      final ||
+        ((FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_CHECKPOINT) ||
+           FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA)) &&
+          !final));
```
Comment (keitharnoldsmith) [inline src/txn/txn_ckpt.c:1960]: The `!final` at the end of the assert is unnecessary. You can remove it.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

One comment, below, about simplifying the assert.  No need to have me re-review the change.

File: src/txn/txn_ckpt.c [hunk 2]
```diff
@@ -1948,6 +1948,16 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
         (!F_ISSET(S2C(session), WT_CONN_FILE_CLOSE_SYNC) && !metadata)))
         return (__wt_set_return(session, EBUSY));
 
+    /*
+     * Checkpoint shouldn't occur when final is set to false, without either checkpoint lock or
+     * schema lock. Potential race condition arises when backup is ongoing, causing inconsistencies
+     * between the versions of data.
```
Comment (sueloverso) [inline src/txn/txn_ckpt.c:1954]: The assert itself is fine. I think the comment needs more information. Maybe replace with something like:
```
/*
 * Make sure there isn't a potential race between backup copying the metadata and a checkpoint changing the metadata.
 * Backup holds both the checkpoint and schema locks. Checkpoint should hold those also except on the final checkpoint
 * during close. Confirm the caller either is the final checkpoint or holds at least one of the locks.
 * /
 ```

**Review by sueloverso — APPROVED** [review]
LGTM with a suggestion for comment wording.

**Pull request merged**